### PR TITLE
cmd/evm: add blocktest subcommand to evm

### PR DESCRIPTION
**Pull request by marioevz**
Adds `blocktest` subcommand to the `evm` command, which is very similar to `statetest`, but instead of loading a StateTest static test it loads a BlockchainTest from a json file and runs it.

I use this a lot to double check tests are filled correctly on https://github.com/ethereum/execution-spec-tests/, and normally when testing a change I cherry-pick this commit on top of whatever I'm testing, but it might be useful to have it in master.

### CODE REVIEW

**Review by lightclient — APPROVED** [review]
LGTM

File: cmd/evm/blockrunner.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	var tests map[string]tests.BlockTest
+	if err = json.Unmarshal(src, &tests); err != nil {
+		return err
+	}
+	for _, test := range tests {
+		if err := test.Run(false); err != nil {
+			return err
+		}
+
```
Comment (lightclient) [inline cmd/evm/blockrunner.go:70]:
```suggestion
```

**Review by lightclient — COMMENTED** [review]
this is actually blocking CI

File: cmd/evm/blockrunner.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// BlocktestResult contains the execution status after running a blockchain test, any
+// error that might have occurred and a dump of the final blockchain if requested.
+type BlocktestResult struct {
+	Name  string      `json:"name"`
+	Pass  bool        `json:"pass"`
+	Fork  string      `json:"fork"`
+	Error string      `json:"error,omitempty"`
+	State *state.Dump `json:"state,omitempty"`
+}
```
Comment (holiman) [inline cmd/evm/blockrunner.go:46]: I suppose the intention was to actually use this? 
Right now, it's totally silent if the test passes
```
$ ./evm blocktest /home/user/go/src/github.com/ethereum/go-ethereum/tests/testdata/BlockchainTests/GeneralStateTests/stChainId/chainId.json 
$
```

File: cmd/evm/blockrunner.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/tests"
+
```
Comment (holiman) [inline cmd/evm/blockrunner.go:27]:
```suggestion
```

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**